server/middleware/auth.js

```diff
@@ -26,6 +26,23 @@ function authenticateJWT(req, res, next) {
 	}
 }
 
+/**Middleware: Socket Authentication */
+function socketAuth(socket, next) {
+	const token = socket.handshake.auth.token;
```

Comment:
You should be able to send and parse the data as a cookie just like with a regular HTTP request. The cookie string is available as `socket.request.headers.cookie`, and you can parse it using a third party library like [cookie-parser](https://www.npmjs.com/package/cookie-parser), which is included as part of the dependencies for this project.

Reply:
Okay, i will make the update to cookies. One of the posts i read was saying that Auth is the perferred over cookies to send credentials in socket.io

Reply:
The only issue is that if you're sending the same token you're sending via the cookies, that makes using the cookies pointless.

Reply:
@tmattmac I'm just looking at this now. Having a think about how to move this into cookies. I'm having an issue where the socket connection isn't occurring until the HTTP Auth process is finished. Given this the server doesn't know who the user is & requires for the user to provide some context (i was using the token for this for some protection). 

By moving to cookies the server is still going to need some information of who the user is when setting up the connection, would it be acceptable to use a token but with a separate secret key? Right now we know that the user has to be authenticated with HTTP-only cookies to be able to access the SocketProvider (otherwise they get re-routed to the auth page).

The other thing i was thinking of was to move the socketProvider outside all routes & we lose the protection of knowing that the user has passed the http authentication process and would need http-only cookies on the socket connection. The issue i then have is how to know the Socket ID when the user logs in so that i can add the cookies to only that socket.

Reply:
It's generally not a good idea to rely on client-side routing as a security guarantee.

The token stored in the cookies should still be able to provide the user information to the socket. Is there any reason it wouldn't?

Reply:
Plus, it's best to stick to a single source of truth for whether the user is logged in, so it'd be preferred to stick with one token.

Reply:
@tmattmac With the current set up the cookie isn't attached to the socket. I'll take a look at the setup to see what needs to be amended to update this.

Reply:
I think it might be related to CORS. You should be able to add a [`withCredentials`](https://socket.io/docs/v3/handling-cors/) option when initializing the connection via the client. Alternatively, I think you could connect directly via the proxy rather than through the API URL, if you don't plan on deploying the server and client code to separate servers.

Reply:
Thanks Trey, i was just about to saw i found out the issue and you're 100% right i just needed to allow credentials to the CORS. options. I'm updating the code now to remove the previous auth and will let you know once the update has been pushed up.